### PR TITLE
[Model] Officially support Emu3 with Transformers backend

### LINKED ISSUE #11008: [New Model]: Emu3
### The model to consider.

https://huggingface.co/BAAI/Emu3-Chat

### The closest model vllm already supports.

_No response_

### What's your difficulty of supporting the model you want?

i want to know when will support Emu3 model.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by hmellor**
This PR adds an explicit redirect from `Emu3ForConditionalGeneration` to `TransformersForMultimodalLM`. This means that:

- The Transformers backend will be used to load this model
- Users won't see warnings about the Transformers backend being fallback behaviour

In the docs I have added a paragraph explaining what it means for a model to be officially supported via the Transformers backend.

Closes #11008

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]
Thanks, hope that this would encourage more people to consider Transformers backend before opening their own PRs!

**Comment by DarkLight1337** [comment]
PTAL at multi-modal tests failure

**Comment by hmellor** [comment]
It's a weight mapping issue. We've had a TODO in the code for a while to simplify the weight mapping so I've made https://github.com/vllm-project/vllm/pull/21382 to do that ahead of this PR.

**Review by DarkLight1337 — APPROVED** [review]

**Comment by hmellor** [comment]
Fixing the `hf_to_vllm_mapper` revealed an issue where the running stats are not being loaded into the `nn.BatchNorm3d` in Emu3's vision tower. Currently working on a fix.

**Comment by hmellor** [comment]
The failing mapper test was using `model.named_parameters()` as reference but the Emu3 checkpoint also contains buffers (stats for batch norm).

I have modified the test to ignore buffers that it may find in checkpoints.
